rfcs/2021-19-02-units-for-typography-and-layout.md

```diff
@@ -0,0 +1,603 @@
+<img alt="Layout px text rem" src="./images/unit-font-size-rem-zoom.png" width="400px">
+
+#### Output - `Layout absolute units, typography relative unit` with default font-size of 32px <!-- omit in toc -->
+<img alt="Layout absolute units, typography relative unit" src="./images/unit-layout-absolute-font-relative.png" width="800px">
+
+#### Output - `Layout relative units, typography relative unit` with default font-size of 32px <!-- omit in toc -->
+<img alt="Layout relative units, typography relative unit" src="./images/unit-layout-relative-font-relative.png" width="800px">
+Can you spot the difference when everything is relative unit? The layout also changed with the increase in font size and it looks different from what we expected it to look like.
+<br/> <br/>
+
+> 🔗 You can play around with all the [demos here](https://szi8i.csb.app/)
+
+#### For React Native <!-- omit in toc -->
+It's **layout in dp and typography in dp w/ autoscale**. Why?
+* React native platform only allows `dp` as units.
+* It's better since the layouts become much more predictable as the platform just asks us to define things at a base layout and takes care of scaling on different devices with different screen densities automatically.
+* We will always honor user's device font size since we want the content to be readable as per user's preference. It might break the layout in some cases but the cons weighs off the pros i.e **content readability**
+
+Let's tally our decision with our [checklist](#checklist)
+1. Whether the content is accessible when the user changes the default font size of the browser/device ✅
+2. Whether the content is accessible when the user zooms in/out on the browser ✅
+3. Whether the interface is able to render on different screen sizes, pixel densities etc ✅
+
+
+## How will we store and use it?
+<img alt="storing and getting tokens" src="./images/unit-storing-and-getting.png" width="1200px">
+
+* While storing we can compute the value in `px/dp` and store it in a unitless way. While rendering we can attach the units.
+   ```js
+   const space = {
+      1: 8,
+      2: 12,
+      3: 16,
+   }
+   const fontSize = {
+      1: 14,
+      2: 18,
+      3: 24,
+   }
+   ```
+* With the above approach we can keep the vocabulary consistent that everything is stored in `px` for web and `dp` for react native  but while rendering the respective platforms(web/react-native) attaches the unit for the target rendering engine.
+* We will implement a generic funtion that will attach the units to the tokens.
+* For typography the units that'll be constructed will be relative(`rems` for web and `autoScale` value on text enabled for react-native apps).
+* For layout i.e height, width, padding, margin the units that'll be constructed will be absolute i.e `px`(pixels) for web and `dp` for react-native apps but with device pixel ratio to ensure consistency with various screen densities.
```

Comment:
> `dp` for react-native apps but with device pixel ratio to ensure consistency with various screen densities.

I though we are just attaching `dp` and not taking `PixelRatio` into account anymore based on https://github.com/razorpay/blade/pull/246/files#diff-e1764707dceb6b254a8543b8f0578996940d0001820faea450cd29911623f110R520.

Reply:
Good catch because of back and forth I forgot to change this based on final conclusion 😄